src/vs/workbench/contrib/chat/browser/chatEditingActions.ts

```diff
@@ -273,3 +279,45 @@ registerAction2(class AddFilesToWorkingSetAction extends Action2 {
 		}
 	}
 });
+
+registerAction2(class CancelAction extends Action2 {
+	static readonly ID = 'workbench.action.chat.editing.cancel';
+	constructor() {
+		super({
+			id: CancelAction.ID,
+			title: localize2('workbench.action.chat.editing.cancel.label', "Cancel"),
+			f1: false,
+			category: CHAT_CATEGORY,
+			icon: Codicon.stopCircle,
+			menu: {
+				id: MenuId.ChatExecute,
+				when: ContextKeyExpr.and(CONTEXT_CHAT_LOCATION.isEqualTo(ChatAgentLocation.EditingSession), ContextKeyExpr.or(CONTEXT_CHAT_REQUEST_IN_PROGRESS, applyingChatEditsContextKey)),
```

Comment:
If it makes things simpler, you could also just keep one command that does something different in this state. It would mean that as a user I don't have to customize keybindings for two commands for example.

Reply:
https://github.com/microsoft/vscode/pull/230743, I was worried about accidentally canceling edits if the user has both views open simultaneously, but as long as action context is properly set that shouldn't be an issue